### PR TITLE
Update from Upstream

### PR DESCRIPTION
**Pull request by claravanstaden**
Updates snowbridge with latest changes from the upstream polkadot-sdk.

Resolves: [SNO-764](https://linear.app/snowfork/issue/SNO-764)

### CODE REVIEW

File: cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs [hunk 1]
```diff
@@ -489,6 +460,25 @@ impl<
...
+			return false
+		}
+		WaivedLocations::contains(loc)
+	}
+
+	fn handle_fee(fee: MultiAssets, context: Option<&XcmContext>, reason: FeeReason) {
+		FeeHandler::handle_fee(fee, context, reason);
+	}
+}
+
```
Comment (claravanstaden) [inline cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs:481]: In this latest version, fees from asset hub to bridgehub are waived. There was a bug in the previous version, which is why it is only failing now. Introduced `XcmFeeManagerFromComponentsBridgeHub` to check if the fee reason is `Export` and from the Ethereum network. If so, a fee needs to be paid.

Let me know if you think this is OK and if I should check if the origin is from Asset hub.

**Review by alistair-singh — APPROVED** [review]
LGTM! 🚀

**Review by yrong — APPROVED** [review]
Cool!

**Review by vgeddes — APPROVED** [review]
Looks good

**Pull request merged**